src/mjolnir/graphenhancer.cc

```diff
@@ -1074,6 +1075,10 @@ uint32_t GetStopImpact(uint32_t from,
     } else if (stop_impact != 0) { // make sure we do not subtract 1 from 0
       stop_impact -= 1;
     }
+  } else if (edges[from].use() == Use::kParkingAisle && edges[to].use() == Use::kParkingAisle) {
+    // decrease stop impact inside parking lots
+    if (stop_impact != 0)
+      stop_impact -= 1;
   }
```

Comment:
Pattern `if (stop_impact != 0)  stop_impact -= 1;` is used often in the code. Maybe introduce a lambda like `auto decrease_stop_impact = [&stop_impact](){if (stop_impact != 0)  stop_impact -= 1;}`?

Reply:
@averkhaturau How about 
```
auto decrement_if_not_zero = [](uint32_t& stop_impact){if (stop_impact != 0) stop_impact -= 1;};
```
I don't quite like the initial suggestion in terms of readability. Additionally I'd rather not capture local value by reference. I understand it's ok in this very case but still don't like such approach

Reply:
I'm ok with repeating the original code, imho it doesn't worth to make a general purpose solution for a trivial tasks.